### PR TITLE
[Feature] Rehire employee process

### LINKED ISSUE #695: [Feature] Rehire employee process
Similar to onboarding, we should have a rehire process. Maybe it's just kicking the record into onboarding??

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #695
close with associatedemployers/igneous#305

image

image

image

### CODE REVIEW

File: app/controllers/account/employee/index.js [hunk 1]
```diff
@@ -20,6 +21,30 @@ export default Controller.extend(del, {
         this.ajaxSuccess('Email sent.');
       })
       .catch(this.ajaxError.bind(this));
+    },
+
+    async rehire () {
+      this.model.eventType.push({
```
Comment (James1x0) [inline app/controllers/account/employee/index.js:27]:
```suggestion
      this.model.eventHistory.push({
```

File: app/controllers/account/employee/index.js [hunk 2]
```diff
@@ -20,6 +21,30 @@ export default Controller.extend(del, {
...
+        termination: this.model.terminatedOn,
+        rehired:     moment().toDate(),
```
Comment (James1x0) [inline app/controllers/account/employee/index.js:29]:
```suggestion
        rehired:     new Date(),
```

File: app/controllers/account/employee/index/index.js [hunk 3]
```diff
@@ -12,6 +14,16 @@ export default Controller.extend(addEdit, {
...
 
+  allNotes: computed('model.eventType.@each.notes', function () {
+    return this.get('model.eventType').reduce((arr, event)=>{
+      arr.push({
+        notes: event.notes,
+        date:  moment(event.termination).format('M/D/YYYY')
+      });
+      return arr;
+    }, []);
+  }),
```
Comment (James1x0) [inline app/controllers/account/employee/index/index.js:25]: This is pretty odd... I don't think this should be a function, you can probably do all of this in the template

File: app/controllers/account/employee/index/index.js [hunk 4]
```diff
@@ -55,6 +67,10 @@ export default Controller.extend(addEdit, {
 
       model.set('customFields', customFields);
       this.saveModel();
+    },
+
+    openModal () {
```
Comment (James1x0) [inline app/controllers/account/employee/index/index.js:72]: We are in a complicated route, give this function name something more descriptive.

File: app/models/employee.js [hunk 5]
```diff
@@ -40,6 +40,8 @@ export default Model.extend(Validations, {
   workersCompClassCode:      attr('string'),
   employeeNumber:            attr('string'),
 
+  eventType: attr('array'),
```
Comment (James1x0) [inline app/models/employee.js:43]:
```suggestion
  eventHistory: attr('array'),
```

File: app/templates/account/employee/index/index.hbs [hunk 6]
```diff
@@ -216,7 +216,32 @@
       {{/link-to}}
       <div class="ui basic segment">
         <p class="text">
-          {{#if (or this.model.probationUntil this.model.ssn)}}
+          {{#if (or this.model.probationUntil this.model.ssn this.model.eventType) }}
```
Comment (James1x0) [inline app/templates/account/employee/index/index.hbs:219]:
```suggestion
          {{#if (or this.model.probationUntil this.model.ssn this.model.eventHistory.length) }}
```

File: app/templates/account/employee/index/index.hbs [hunk 7]
```diff
@@ -216,7 +216,32 @@
...
+              </div>
+            {{/ui-modal}}
+            {{#if this.allNotes}}
+              <div class="line-item-value">
+                <i>
+                  Notes
+                </i>
+                {{ember-tooltip side='right' text='view previous separation notes'}}
+                <u>
+                  <a href="#" class="right floated" {{action "openModal"}}>More...</a>
```
Comment (James1x0) [inline app/templates/account/employee/index/index.hbs:241]:
```suggestion
                  <a href="#" class="right floated" {{action "openModal"}}>View</a>
```

File: app/templates/account/employee/index/index.hbs [hunk 8]
```diff
@@ -216,7 +216,32 @@
...
+                  </div>
+                {{/each}}
+              </div>
+            {{/ui-modal}}
+            {{#if this.allNotes}}
+              <div class="line-item-value">
+                <i>
+                  Notes
+                </i>
+                {{ember-tooltip side='right' text='view previous separation notes'}}
```
Comment (James1x0) [inline app/templates/account/employee/index/index.hbs:239]:
```suggestion
                {{ember-tooltip side='right' text='View all notes'}}
```

File: app/templates/components/list-item/history-item.hbs [hunk 9]
```diff
@@ -6,11 +6,17 @@
...
 <div class="ui divided list">
   {{#each this.history.diff as |diff|}}
-    <div class="item">
-      <strong>{{humanize-key (join-array diff.path ".")}}</strong>
-      changed from
-      <span class="ui label">{{diff-value-translator key=diff.path value=diff.lhs}}</span> to
-      <span class="ui green label">{{diff-value-translator key=diff.path value=diff.rhs}}</span>
-    </div>
+    {{#if diff.item.rhs.rehired}}
+      <strong>Rehired</strong>
```
Comment (James1x0) [inline app/templates/components/list-item/history-item.hbs:10]: Put this in an "item"
Lists should all have "item" children

File: app/controllers/account/employee/index/index.js [hunk 10]
```diff
@@ -2,6 +2,8 @@ import Controller from 'granite/core/controller';
 import { computed, action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
 import { inject as service } from '@ember/service';
+import $ from 'jquery';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (JavaLavaMT) [reply app/controllers/account/employee/index/index.js:6]: https://github.com/associatedemployers/granite/blob/feature%2F%23695-rehire-employee-process/app/controllers/account/employee/index/index.js#L24

[the remark this answers is not in the document]
Comment (James1x0) [reply app/controllers/account/employee/index/index.js:6]: That’s weird why does it show you’re adding this line??!

[the remark this answers is not in the document]
Comment (JavaLavaMT) [reply app/controllers/account/employee/index/index.js:6]: I think I added jquery and maybe deleted moment and brought it back on a new line after jquery haha weird

**Pull request merged**